packages/core/src/webviews/main.ts

```diff
@@ -225,12 +239,38 @@ export abstract class VueWebview {
             }
 
             public async setup(webview: vscode.Webview) {
+                this.setupTelemetry()
+
                 const server = registerWebviewServer(webview, this.instance.protocol, this.instance.id)
                 this.instance.onDidDispose(() => {
                     server.dispose()
                 })
             }
 
+            /**
+             * Setup telemetry events that report on the initial loading of a webview
+             */
+            private setupTelemetry() {
+                this.instance.traceId = randomUUID()
+                // Notify intent to open a module, this does not mean it successfully opened
+                telemetry.toolkit_willOpenModule.emit({
```

Comment:
Is there a case where we intend to open a module (emit `willOpenModule`) then don't also emit a `didLoadModule` with either fail or succeed?

Reply:
For now, every webview will emit a `willOpenModule` and `didLoadModule` will need to be explicitly done by each webview.

This is due to how loading a webview works. We can only indicate our intent to open a webview but have no formal way to know when it has opened (we create a vscode webview instance and set a string of HTML, then have no insight to what happens after that)

So there will be cases where we only have `willOpenModule` and no trailing `didLoadModule`. This is essentially how we have it right now, but under different metric names.

Reply:
In most cases, did/will pairs should not be needed. Only the "did" case is needed for most metrics, because the metric will wrap the impl logic and track a `duration`. Why do we need both here?